### ISSUE TITLE
nvcc fatal : Unknown option 'fopenmp' using CMAKE

### ISSUE THREAD

**Issue by adujardin** [AlexeyAB/darknet#3314]
Hi, 
Using the vanilla CMakeLists to compile I have the following issue on Ubuntu 18 (default cmake 3.10.2, g++ 7.4), CUDA 10 :

`nvcc fatal   : Unknown option 'fopenmp'`

I tried several fixes, including replacing `--compiler-options` by `-Xcompiler` as a common fix for this issue with no luck, here:
https://github.com/AlexeyAB/darknet/blob/5b8929becf23496666788d86450d2a47d6745ef0/CMakeLists.txt#L171

One solution found was to comment these lines: https://github.com/AlexeyAB/darknet/blob/5b8929becf23496666788d86450d2a47d6745ef0/CMakeLists.txt#L306-L307

@cenit Do you have an idea why nvcc complains about that?
Are these 2 lines necessary?

**Comment by cenit** [comment]
Hi @adujardin 
let's fix this problem. First of all do you have the same symptoms using #2952? I fixed CMakeLists.txt a little bit there, I am waiting on @AlexeyAB for some feedback for merge but otherwise it should be production ready

**Comment by adujardin** [comment]
Hi @cenit 
I tried your branch, however by default CUDA detection have a weird behavior:

**1.** Cuda detection with default CMakeLists

```
[...]
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - /usr/local/cuda/bin/nvcc
-- The CUDA compiler identification is NVIDIA 10.0.130
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc -- works
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
[...]
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find CUDA (missing: CUDA_INCLUDE_DIRS) (found version "10.0")
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.10/Modules/FindCUDA.cmake:1080 (find_package_handle_standard_args)
  CMakeLists.txt:51 (find_package)


-- Configuring incomplete, errors occurred!
```
When poking around [l.51](https://github.com/cenit/darknet/blob/dev/cenit/include/CMakeLists.txt#L51), the cmake process either fail or the compilation works but without CUDA.

The cmake log only report a pthread issue which seems unrelated :confused: 

[CMakeError.log](https://github.com/AlexeyAB/darknet/files/3252238/CMakeError.log)
[CMakeOutput.log](https://github.com/AlexeyAB/darknet/files/3252239/CMakeOutput.log)

It could be related to the cmake version which is semi-old. But ideally, I think there should be a workaround for this version since it's the default on the latest Ubuntu LTS.

**2.** I then tried to force CUDA by removing every "REQUIRED" from `find_package(CUDA)` (l.51) + commenting ZED Camera support which also include another find_package CUDA, I managed to get the **same issue** `nvcc fatal   : Unknown option 'fopenmp'`

**3.** `find_package(CUDNN)` doesn't exist since cuDNN was installed by simply copying the library and headers. Is there a standard way to get this find package file?

--

Do you know if there's a way to check if a library have been compiled with OpenMP? I would like to check if removing these makes a difference on the final lib/exe. https://github.com/AlexeyAB/darknet/blob/5b8929becf23496666788d86450d2a47d6745ef0/CMakeLists.txt#L306-L307

**Comment by adujardin** [comment]
Just in case I also tried with the [docker image provided by nvidia](https://hub.docker.com/r/nvidia/cuda) matching my configuration to have a clean setup.

I don't have the issues mentioned above, except the `nvcc fatal : Unknown option 'fopenmp'`.
You can ignore those, I'll try to investigate why since I installed it using the standard way.

**Comment by cenit** [comment]
A proper fix is contained in #3371
